src/plugins/github/fetchGithubRepo.js

```diff
@@ -49,6 +50,75 @@ export default function fetchGithubRepo(
 
 const GITHUB_GRAPHQL_SERVER = "https://api.github.com/graphql";
 
+type GithubResponseError =
+  | {|+type: "FETCH_ERROR", retry: false, error: Error|}
+  | {|+type: "GRAPHQL_ERROR", retry: false, error: mixed|}
+  | {|+type: "GITHUB_INTERNAL_EXECUTION_ERROR", retry: true, error: mixed|}
+  | {|+type: "NO_DATA", retry: true, error: mixed|};
+
+// Fetch against the GitHub API with the provided options, returning a
+// promise that either resolves to the GraphQL result data or rejects
+// to a `GithubResponseError`.
+function tryGithubFetch(fetch, fetchOptions): Promise<any> {
+  return fetch(GITHUB_GRAPHQL_SERVER, fetchOptions).then(
+    (x) =>
+      x.json().then((x) => {
+        if (x.errors) {
+          if (
+            x.errors.length === 1 &&
+            x.errors[0].message.includes("it could be a GitHub bug")
+          ) {
+            return Promise.reject(
+              ({
+                type: "GITHUB_INTERNAL_EXECUTION_ERROR",
+                retry: true,
+                error: x,
+              }: GithubResponseError)
+            );
+          } else {
+            return Promise.reject(
+              ({
+                type: "GRAPHQL_ERROR",
```

Comment:
Hmm, so if GitHub changed the message they put on their timeout failure, we'll start registering them as GRAPHQL_ERRORs. in that case, it would reject without retry and the error would be printed to console, and we would quickly see what went wrong. so I think this is fine.

Reply:
Precisely correct. I agree that this is fine. If we start getting
spurious errors for whatever reason, we can of course revisit this
logic.